### PR TITLE
Run cppcheck

### PR DESCRIPTION
**Pull request by mikeessen**
Invoke cppcheck to perform cppcheck scan.

### CODE REVIEW

File: docker/Dockerfile.onnx-mlir-dev [hunk 1]
```diff
@@ -48,6 +48,10 @@ RUN LLVM_PROJECT_ROOT=${WORK_DIR}/llvm-project \
             -j${NPROC} \
             check-onnx-backend-numerical \
     && make check-docs \
+# Run the cppcheck scanner on s390x
+    && if [ "$(uname -m)" = "s390x" ]; then \
+           ${ONNX_MLIR_ROOT}/utils/cppcheck_scan/cppcheck-scan.py; \
+       fi \
     && rm -f Debug/bin/*Test Debug/bin/Perf* Debug/bin/Test* \
```
Comment (gongsu832) [inline docker/Dockerfile.onnx-mlir-dev:55]: cppcheck is a static check. It should be run as early as possible. No point in running all the builds and tests successfully only later failing the cppcheck. I would move it up and have a separate RUN for it, right after
```
# Copy onnx-mlir from the pull request
COPY . onnx-mlir
```

Comment (mikeessen) [reply docker/Dockerfile.onnx-mlir-dev]: The cppcheck is being invoked with the project file from cmake - `onnx-mlir/build/compile_commands.json`.  Moving the cppcheck up to after that file is created.

File: docker/Dockerfile.llvm-project [hunk 2]
```diff
@@ -34,7 +34,10 @@ RUN distro=$(cat /etc/os-release|grep -Po '(?<=^ID=").*(?=")|(?<=^ID=)[^"].*[^"]
                   python3 python3-dev python3-distutils python3-numpy \
                   python3-pip python3-pytest-xdist python3-setuptools \
                   python3-typing-extensions zlib1g-dev && \
-          rm -rf /var/lib/apt/lists/* && \
+          if [ "$(uname -m)" = "s390x" ]; then \
+              apt-get install -qq -y --no-install-recommends cppcheck; \
+          fi \
+          && rm -rf /var/lib/apt/lists/* && \
           ln -sf /usr/bin/pytest-3 /usr/bin/pytest; \
```
[the remark this answers is not in the document]
Comment (mikeessen) [reply docker/Dockerfile.llvm-project]: Added cppcheck to the prior list for apt-get.  Thanks!

**Comment by gongsu832** [comment]
Can we make cppcheck less verbose when there is no error and only prints out error messages? Otherwise the console output gets cluttered with hundreds of 
```
Checking /workdir/onnx-mlir/third_party/onnx/onnx/checker.cc ...
Checking /workdir/onnx-mlir/third_party/onnx/onnx/checker.cc: ONNX_ML=1;ONNX_NAMESPACE=onnx;_DEBUG=1;_GLIBCXX_ASSERTIONS=1;_LIBCPP_ENABLE_ASSERTIONS=1;__STDC_CONSTANT_MACROS=1;__STDC_FORMAT_MACROS=1;__STDC_LIMIT_MACROS=1;__PIC__=1;__PIC__=1;_DEBUG=1;_GLIBCXX_ASSERTIONS=1;_LIBCPP_ENABLE_ASSERTIONS=1;__STDC_CONSTANT_MACROS=1;__STDC_FORMAT_MACROS=1;__STDC_LIMIT_MACROS=1;__cplusplus=201103L...
1/639 files checked 0% done
Checking /workdir/onnx-mlir/third_party/onnx/onnx/common/assertions.cc ...
Checking /workdir/onnx-mlir/third_party/onnx/onnx/common/assertions.cc: ONNX_ML=1;ONNX_NAMESPACE=onnx;_DEBUG=1;_GLIBCXX_ASSERTIONS=1;_LIBCPP_ENABLE_ASSERTIONS=1;__STDC_CONSTANT_MACROS=1;__STDC_FORMAT_MACROS=1;__STDC_LIMIT_MACROS=1;__PIC__=1;__PIC__=1;_DEBUG=1;_GLIBCXX_ASSERTIONS=1;_LIBCPP_ENABLE_ASSERTIONS=1;__STDC_CONSTANT_MACROS=1;__STDC_FORMAT_MACROS=1;__STDC_LIMIT_MACROS=1;__cplusplus=201103L...
2/639 files checked 0% done
...
```

**Review by gongsu832 — APPROVED** [review]
LGTM

**Pull request merged**